src/app/data-core/ICourse.ts

```diff
@@ -0,0 +1,7 @@
+export interface ICourse {

```

Comment:
I  think it's better to move interfaces to separate folder `interfaces`. `data-core` than can be renamed to something like `models`.
Also you can review great official style guide https://angular.io/guide/styleguide

Reply:
Changed